packages/realm-network-transport/src/NetworkTransport.ts

```diff
@@ -1,7 +1,7 @@
 declare var process: any;
 declare var require: ((id: string) => any) | undefined;
 
```

Comment:
> line 19: shouldn't body in the Response be optional?

No - the body of the request is decoded using fetch's `text()` method, which always returns a (possibly empty) string.

> line 53 and 71: probably make this as a isNodeProcess constant and have it declared at the beginning

Makes sense - I'll leave the `typeof require === "function"` check, since that helps type inference allowing me to call the require function.

> line 88-91 formatting should be on single line

This will change once we merge https://github.com/realm/realm-js/pull/2794 and run the `--fix`. I simply forgot to add `printWidth: 120` to the prettierrc.js added with https://github.com/realm/realm-js/pull/2811.

> why not implement fetchWithCallbacks as an async function. I think we should by default try using async functions to be able to await which will give us the benefit of that syntax over Promise.then().catch()

Because this is exactly the intent of `fetchWithCallbacks`: To provide a non-promise-based API to the fetch implementation. This was a requirement from @kneth as the code is called by C++ where it's non-trivial to call `then` on the promise returned.